lib/TokenManager.ts

```diff
@@ -287,9 +287,28 @@ export class TokenManager {
     this.off = tokenMgmtRef.emitter.off.bind(tokenMgmtRef.emitter);
     this.hasExpired = hasExpired.bind(this, tokenMgmtRef);
   
+    let tooManyRenews = false;
+    const renewTimeQueue = [];
     const onTokenExpiredHandler = (key) => {
       if (options.autoRenew) {
-        this.renew(key).catch(() => {}); // Renew errors will emit an "error" event 
+        // detect if there are too many renew requests
```

Comment:
This comment doesn't really line up with what follows

Reply:
can you explain more on how it's not lined up? it's logic to assign value to the `tooManyRenews` flag, which is the rate-limiting detection here.

Reply:
It looks like it's describing the 'if' statement, but the if condition is actually "not yet too many".

Reply:
I'll extract it to an isolated function, should be more declarative.